tests/test_qc.py

```diff
@@ -247,9 +247,14 @@ def test_go_rules_15():
     assert test_result.result_type == qc.ResultType.WARNING    
 
 def test_go_rule_16():
+    # GO term same as with/ID
+    assoc = make_annotation(goid="GO:0044419", evidence="IC", withfrom="GO:0044419").associations[0]
```

Comment:
This test is good and I think another good test would be to include multiple GO terms in the test `withfrom`, one valid (differs from `goid`) and one invalid (same) to ensure it's not dropping `withfrom` entirely. So:
```
assoc = make_annotation(goid="GO:0044419", evidence="IC", withfrom="GO:0044419|GO:0035821").associations[0]
```
This should return `qc.ResultType.PASS` because the with/from still has >=1 valid GO term.

Reply:
Good point.  Also code should output message even if one of the terms is same as go id for annotation